plugins/readme/girder_readme/rest.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+import re
+import cherrypy
+from girder.api import access
+from girder.api.describe import Description, autoDescribeRoute
+from girder.constants import AccessType, TokenScope
+from girder.models.file import File as FileModel
+from girder.models.folder import Folder as FolderModel
+from girder.models.item import Item as ItemModel
+
+
+@access.public(scope=TokenScope.DATA_READ)
+@autoDescribeRoute(
+    Description('Get the README for a folder, if it exists.')
+    .modelParam('id', model=FolderModel, level=AccessType.READ)
+    .errorResponse()
+    .errorResponse('Read access was denied on the folder.', 403)
+)
+def _getFolderReadme(folder):
+    query = {
+        'folderId': folder['_id'],
+        'lowerName': {'$regex': re.compile('^readme')},
```

Comment:
We have a compound index for `folderId` and `name`, but not for `folderId` and `lowerName`. So let's search using `name`, and maybe for now we require all-caps for the README part of the name.

We could also refine this regex to be a little more specific e.g. `^README$|^README\..*`